### PR TITLE
Pangram: Add Pangram exercise

### PR DESCRIPTION
**Pull request by Akasurde**
This PR adds exercise for determining if a sentence
is a pangram or not.

Fixes https://github.com/exercism/xgo/issues/395

Signed-off-by: Abhijeet Kasurde akasurde@redhat.com

### CODE REVIEW

File: exercises/pangram/example.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package pangram
+
+import (
+	"strings"
+)
+
+const testVersion = 1
+
+func is_pangram(s string) bool {
```
Comment (kytrinyx) [inline exercises/pangram/example.go]: Go uses MixedCaps for names: https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps

File: exercises/pangram/pangram_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	input    string
+	expected bool
+}
+
+var stringTestCases = []testCase{
+	{"quick movement of the enemy will jeopardize five gunboats", false},
+	{"the quick brown fish jumps over the lazy dog", false},
+	{"7h3 qu1ck brown fox jumps ov3r 7h3 lazy dog", false},
+	{"the 1 quick brown fox jumps over the 2 lazy dogs", true},
+	{"The quick brown fox jumps over the lazy dog", true},
```
Comment (kytrinyx) [inline exercises/pangram/pangram_test.go]: We have a canonical set of test data that we encourage all of the tracks to use: https://github.com/exercism/x-common/blob/master/exercises/pangram/canonical-data.json

If you see any edge cases that you've included and the canonical json is missing, please open an issue in https://github.com/exercism/x-common/issues so that we can discuss it.

File: exercises/pangram/pangram_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
+	}
+}
+
+func TestPangram(t *testing.T) {
+	for _, test := range stringTestCases {
+		actual := is_pangram(test.input)
+		if actual != test.expected {
+			t.Errorf("Pangram test [%s], expected [%s], actual [%s]", test.input, test.expected, actual)
```
Comment (kytrinyx) [inline exercises/pangram/pangram_test.go]: One of the things we try to do is help people understand _why_ a test fails. How about adding a _description_ field to the test case struct that explains the context of the failure (e.g. "missing character $CHAR")

File: exercises/pangram/pangram_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+const targetTestVersion = 1
+
+type testCase struct {
+	input    string
+	expected bool
+}
+
+var stringTestCases = []testCase{
```
Comment (kytrinyx) [inline exercises/pangram/pangram_test.go]: The `string` prefix is unnecessary here. I'd suggest dropping it.

File: exercises/pangram/example.go (hunk 1 above)
Comment (Akasurde) [inline exercises/pangram/example.go]: Ok, I will `is_pangram` to `isPangram`

File: exercises/pangram/pangram_test.go (hunk 4 above)
Comment (Akasurde) [inline exercises/pangram/pangram_test.go]: Ok, I will change `stringTestCases` to `testCases`

File: exercises/pangram/pangram_test.go (hunk 2 above)
Comment (Akasurde) [inline exercises/pangram/pangram_test.go]: OK, I will change dataset to link mentioned.

File: exercises/pangram/pangram_test.go (hunk 3 above)
Comment (Akasurde) [inline exercises/pangram/pangram_test.go]: I am adding `failureReason` in struct and printing the error reason if test is expected to fail.

File: exercises/pangram/pangram_test.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func TestPangram(t *testing.T) {
+	for _, test := range testCases {
+		actual := isPangram(test.input)
+		if actual != test.expected {
+			t.Errorf("Pangram test [%s], expected [%s], actual [%s]", test.input, test.expected, actual)
+		}
+		if !test.expected {
+			fmt.Printf("Pangram test [%s], failure reason : %s\n", test.input, test.failureReason)
```
Comment (Akasurde) [inline exercises/pangram/pangram_test.go]: Not using `t.Logf` as it will require `-v` flag in `go test`. Let me know if it needs to be changed to `t.Logf`.

Comment (petertseng) [inline exercises/pangram/pangram_test.go]: Can you clarify what you intend? Now if the `isPangram` returns `false` when it should return `false`, this will print a `failure` reason even though the test has not failed, is that OK? Seems like it might be confusing to the student. Should this go inside the `if actual != test.expected` instead?

I also believe `Logf` messages *are* shown if tests fail, so it would have been OK to use `Logf`.

File: exercises/pangram/pangram_test.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+func TestTestVersion(t *testing.T) {
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
+	}
+}
+
+func TestPangram(t *testing.T) {
+	for _, test := range testCases {
+		actual := isPangram(test.input)
```
Comment (alebaffa) [inline exercises/pangram/pangram_test.go]: I checked for at least 15 exercises in xgo and they are all exported. I am pretty sure they all are. I think we should continue on this line.

**Comment by petertseng** [comment]
Inactivity: Rebased for config.json, and adding commits to address review comments. Will squash + merge shortly.

**Comment by petertseng** [comment]
Thanks @Akasurde !
